sdk/build-scripts/src/build-package.ts

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { promises as fs, existsSync } from 'fs';
+import * as path from 'path';
+import { build } from 'esbuild';
+import { execSync } from 'child_process';
+
+interface PackageJSON {
+	name: string;
+	exports?: Record<string, string | Record<string, string>>;
+}
+
+const ignorePatterns = [/\.test.ts$/];
+
+buildPackage().catch((error) => {
+	console.error(error);
+	process.exit(1);
+});
+
+async function buildPackage() {
+	const allFiles = await findAllFiles(path.join(process.cwd(), 'src'));
+	const packageJson = await readPackageJson();
+	await clean();
+	await buildCJS(allFiles);
+	await buildESM(allFiles);
+	await buildImportDirectories(packageJson);
+}
+
+async function findAllFiles(dir: string, files: string[] = []) {
```

Comment:
Would it be better to use a glob lib for this? No strong feelings here just curious.

Reply:
not sure what you mean by this?

Reply:
Like you could just use the `glob` npm package and chuck it something like `./**/*.ts` and it'll find all the files for you.

Reply:
🤷 not sure I see the benefit, it's like 5 lines of code.  I generally prefer to avoid deps for simple things.  I think if we got into needing to support something other than everything in `src` that might make sense.  Happy to change it if you feel strongly about it.

Reply:
Yeah that's fine, I don't feel strongly at all was just curious why you went this route.